homeassistant/components/rainmachine/__init__.py

```diff
@@ -207,6 +213,8 @@ async def unpause_watering(call):
     ]:
         hass.services.async_register(DOMAIN, service, method, schema=schema)
 
+    config_entry.add_update_listener(async_reload_entry)
```

Comment:
We should save the unsubscribe callback returned here and call that on entry unload.

https://developers.home-assistant.io/docs/config_entries_options_flow_handler#signal-updates